typescript/sdk/src/core/message.ts

```diff
@@ -111,40 +146,36 @@ export class AbacusMessage {
    * @returns an array of {@link AbacusMessage} objects
    */
   static fromReceipt(
+    multiProvider: MultiProvider,
     core: AbacusCore,
     nameOrDomain: NameOrDomain,
     receipt: TransactionReceipt,
   ): AbacusMessage[] {
     const messages: AbacusMessage[] = [];
     const outbox = new Outbox__factory().interface;
+    const network = resolveDomain(nameOrDomain);
+    const provider = multiProvider.getDomainConnection(network).provider!;
```

Comment:
I feel pretty strongly that the interface to multiprovider needs to allow consumers to specify how to access providers/signers and what to do when it doesnt exist. Happy for it to be out of scope, but IMO very critical in the interface for us to fix asap

Reply:
https://github.com/abacus-network/abacus-monorepo/issues/394